csrc/includes/cpu_adam.h

```diff
@@ -91,11 +91,8 @@ class Adam_Optimizer {
                 __half* dev_params = nullptr);
     inline void IncrementStep(size_t step)
     {
-        if (_step < step) {
+        while (_step < step) {
             _step++;
-            if (_step != step) {

```

Comment:
Could there be non-checkpoint loading scenarios where we would still want to raise this error? Totally makes sense in checkpoint loading we need to align the step count. Just concerned that we’re silently aligning them in all cases now.

Reply:
That's a great point, but since this is coming from Python (https://github.com/microsoft/DeepSpeed/blob/cpu-adam/track-optimizer-step/deepspeed/ops/adam/cpu_adam.py#L119) and is controlled by user, I think we are safe to change this! However, I think this makes sense to print a warning so that the user knows of this update. What do you think?

Reply:
By the way, the current step cannot stay behind the step coming in the cpu-adam step function! I think we should throw a runtime error for that.